### PR TITLE
Introduce hack_and_slash20

### PR DESCRIPTION
**Pull request by sonohoshi**
- Add field: `ApStoneCount:int`
- Add throwing Exception `InvalidRepeatPlayException`, `NotEnoughMaterialException`, `UsageLimitExceedException`
- Remove throwing Exception `PlayCountIsZeroException`

### CODE REVIEW

**Comment by sonohoshi** [comment]
```
[xUnit.net 00:08:36.24]     Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip [FAIL]
  Failed Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip [335 ms]
  Error Message:
   Nekoyume.Action.FailedLoadStateException : [1bB3022899d693ebAe87a42b542b42F656A6AB15, 9c7c438149d4cFD6CA70142393e0F72a78d79b23]Aborted as the game config state was failed to load.
  Stack Trace:
     at Nekoyume.Action.HackAndSlash.Execute(IAccountStateDelta states, Address signer, Int64 blockIndex, IRandom random) in /home/runner/work/lib9c/lib9c/Lib9c/Action/HackAndSlash.cs:line 216
   at Nekoyume.Action.HackAndSlash.Execute(IActionContext context) in /home/runner/work/lib9c/lib9c/Lib9c/Action/HackAndSlash.cs:line 94
   at Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip() in /home/runner/work/lib9c/lib9c/.Lib9c.Tests/Action/Scenario/RuneScenarioTest.cs:line 106
```
I check it

**Comment by sonohoshi** [comment]
> ```
> [xUnit.net 00:08:36.24]     Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip [FAIL]
>   Failed Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip [335 ms]
>   Error Message:
>    Nekoyume.Action.FailedLoadStateException : [1bB3022899d693ebAe87a42b542b42F656A6AB15, 9c7c438149d4cFD6CA70142393e0F72a78d79b23]Aborted as the game config state was failed to load.
>   Stack Trace:
>      at Nekoyume.Action.HackAndSlash.Execute(IAccountStateDelta states, Address signer, Int64 blockIndex, IRandom random) in /home/runner/work/lib9c/lib9c/Lib9c/Action/HackAndSlash.cs:line 216
>    at Nekoyume.Action.HackAndSlash.Execute(IActionContext context) in /home/runner/work/lib9c/lib9c/Lib9c/Action/HackAndSlash.cs:line 94
>    at Lib9c.Tests.Action.Scenario.RuneScenarioTest.Craft_And_Equip() in /home/runner/work/lib9c/lib9c/.Lib9c.Tests/Action/Scenario/RuneScenarioTest.cs:line 106
> ```
> 
> I check it

Fixed it

File: Lib9c/Action/HackAndSlash19.cs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+            PlayCount = plainValue["playCount"].ToInteger();
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            if (context.Rehearsal)
+            {
+                return context.PreviousStates;
+            }
+
```
Comment (hskim881028) [inline Lib9c/Action/HackAndSlash19.cs:89]:
```suggestion
CheckObsolete(BlockChain.Policy.BlockPolicySource.V100360ObsoleteIndex, context);
```

**Review by Unengine — APPROVED** [review]

File: Lib9c/Action/HackAndSlash.cs [hunk 2]
```diff
@@ -107,15 +112,25 @@ public override IAccountStateDelta Execute(IActionContext context)
...
-            states.ValidateWorldId(AvatarAddress, WorldId);
+            if (ApStoneCount > UsableApStoneCount)
+            {
+                throw new UsageLimitExceedException(
+                    $"Exceeded the amount of ap stones that can be used " +
+                    $"apStoneCount : {ApStoneCount} > UsableApStoneCount : {UsableApStoneCount}");
+            }
 
-            if (PlayCount <= 0)
+            if (ApStoneCount < 0
```
Comment (ipdae) [inline Lib9c/Action/HackAndSlash.cs:122]: it makes must use ap stone each has action. is this correct?

**Review by hskim881028 — CHANGES_REQUESTED** [review]

**Review by ipdae — APPROVED** [review]

**Pull request merged**